lib/src/robot/client.dart

```diff
@@ -145,19 +151,22 @@ class RobotClient {
         .d('Attempting to reconnect to the robot at $_address every $reconnectInterval ${(reconnectInterval > 1) ? 'seconds' : 'second'}');
 
     while (!_connected) {
+      _sessionsClient.reset();
       try {
-        final channel = await dial(_address, _options);
+        final channel = await dial(_address, _options.dialOptions, () => _sessionsClient.metadata());
         final client = RobotServiceClient(channel);
         await client.resourceNames(ResourceNamesRequest());
 
         _channel = channel;
         _streamManager.channel = _channel as WebRtcClientChannel;
         _client = client;
+        _sessionsClient = SessionsClient(_channel, _options.enableSessions);
         await refresh();
         _connected = true;
         _logger.d('Successfully reconnected robot');
       } catch (e) {
         await _channel.shutdown();
+        _sessionsClient.reset();
```

Comment:
Is this reset needed if we're resetting at the top of the reconnect loop?

Reply:
it may not be, i copied where reset was called in python / TS, so it may be unneeded.